Kinvey/Kinvey/User.swift

```diff
@@ -584,8 +575,10 @@ open class User: NSObject, Credential, Mappable {
         let request = client.networkRequestFactory.buildUserSave(user: self, newPassword: newPassword)
         Promise<U> { fulfill, reject in
             request.execute() { (data, response, error) in
-                if let response = response , response.isOK, let user = client.responseParser.parseUser(data) as? U {
-                    client.activeUser = user
+                if let response = response, response.isOK, let user = client.responseParser.parseUser(data) as? U {
+                    if user.userId == client.activeUser?.userId {
```

Comment:
this is to avoid set the wrong user as an active user if you try to save an user that is not yourself